### PR TITLE
Add docs and import guard to Metaflow integration

### PR DESCRIPTION
**Pull request by andrewtruong**
https://wandb.atlassian.net/browse/WB-6521

Description
-----------

- Add docstrings to user-facing part of integration
- Add import guards for metaflow and fastcore dependencies

Testing
-------

n/a

Release Notes
-------------

------------- BEGIN RELEASE NOTES ------------------

`NO RELEASE NOTES`

------------- END RELEASE NOTES --------------------

[WB-6521]: https://wandb.atlassian.net/browse/WB-6521

### CODE REVIEW

File: wandb/integration/metaflow/metaflow.py [hunk 1]
```diff
@@ -223,6 +235,21 @@ def wandb_log(
...
+    """
+    Automatically log parameters and artifacts to W&B by type dispatch.
+
+    This decorator can be applied to a flow, step, or both.
+    - Decorating a step will enable or disable logging for certain types within that step
+    - Decorating the flow is equivalent to decorating all steps with a default
+    - Decorating a step after decorating the flow will overwrite the flow decoration
+
+    Arguments:
+        datasets (bool): If `True`, log datasets.  Datasets can be a `pd.DataFrame` or `pathlib.Path`.  The default value is `False`, so datasets are not logged.
```
Comment (charlesfrye) [inline wandb/integration/metaflow/metaflow.py:247]: @andrewtruong you'll [need to move the `:` to the other side of the type annotation](https://www.notion.so/wandbai/Docstring-Style-Guide-76d247fc6e0d46a3891bd4c37b3501c1#5c7c84ea12144c9db83ac7ec32146bca) (so `datasets: (bool). If ...`). This is a kind of annoying requirement of our .

You could alternatively add type annotations to the function definition, and then you wouldn't need them here. That's the direction we'd like to move the library. If you go this path, ask someone on client for details about the mypy setup.

**Review by charlesfrye — CHANGES_REQUESTED** [review]
Thanks for adding the import guards!

In addition to resolving the comment, could you add this file to the [docstring style linting white-list](https://www.notion.so/wandbai/Docstring-Style-Guide-76d247fc6e0d46a3891bd4c37b3501c1#ab3f12cbd9914d0c89b0bfc872d8c66d)? LMK if you run into any issues with the process.

File: wandb/integration/metaflow/metaflow.py (hunk 1 above)
Comment (andrewtruong) [reply wandb/integration/metaflow/metaflow.py]: Done

File: .flake8-docstrings [hunk 2]
```diff
@@ -6,4 +6,5 @@ filename =
     ./wandb/sdk/wandb_require.py
     ./wandb/sdk/wandb_run.py
     ./wandb/sdk/wandb_init.py
-    ./wandb/sdk/launch/*
\ No newline at end of file
+    ./wandb/sdk/launch/*
+    ./wandb/integration/metaflow/metaflow.py
```
Comment (raubitsj) [inline .flake8-docstrings:10]: newlines at end of files please

**Review by raubitsj — APPROVED** [review]
Is this ready for merge (it is marked draft)

File: .flake8-docstrings (hunk 2 above)
Comment (andrewtruong) [reply .flake8-docstrings]: Done

Comment (raubitsj) [reply .flake8-docstrings]: Not quite, but it is fine, you have a bad setting in vscode i am guessing that isnt putting a line feed on the last line:
Screen Shot 2021-08-25 at 8 32 56 AM

**Pull request merged**